cli/Dockerfile

```diff
@@ -40,8 +40,8 @@ RUN export DEBIAN_FRONTEND=noninteractive \
      # kubectl CLI
      && apt-get update \
      && mkdir -p /etc/apt/keyrings \
-     && curl -fsSLo /etc/apt/keyrings/kubernetes-archive-keyring.gpg https://dl.k8s.io/apt/doc/apt-key.gpg \
-     && echo "deb [signed-by=/etc/apt/keyrings/kubernetes-archive-keyring.gpg] https://apt.kubernetes.io/ kubernetes-xenial main" | tee /etc/apt/sources.list.d/kubernetes.list \
+     && curl -fsSL https://pkgs.k8s.io/core:/stable:/v1.29/deb/Release.key | gpg --dearmor -o /etc/apt/keyrings/kubernetes-apt-keyring.gpg \
+     && echo 'deb [signed-by=/etc/apt/keyrings/kubernetes-apt-keyring.gpg] https://pkgs.k8s.io/core:/stable:/v1.29/deb/ /' | tee /etc/apt/sources.list.d/kubernetes.list \
```

Comment:
I'm hesitant to using a pinned version here and would prefer we use one of the latest versions mentioned @ https://kubernetes.io/docs/tasks/tools/install-kubectl-linux/#install-using-native-package-management

Reply:
Please see PR #3474 for pin-less strategy.